src/main/java/org/skriptlang/skript/bukkit/recipes/elements/CondSmithingData.java

```diff
@@ -0,0 +1,69 @@
+package org.skriptlang.skript.bukkit.recipes.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.Recipe;
+import org.bukkit.inventory.SmithingRecipe;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Recipe Allows Data Component Copy")
+@Description({
+	"Checks whether the recipe allows data component copying from the base item to the result item of a recipe.",
+	"NOTE: This condition only works on Smithing, Smithing Transform, and Smithing Trim Recipes."
+})
+@Examples({
+	"loop the server's recipes:",
+		"\tif all:",
+			"\t\tthe recipe type of loop-value is a smithing recipe",
+			"\t\tthe loop-value allows data component copying",
+	"then:",
+	"broadcast loop-value"
+})
+@RequiredPlugins("Paper 1.20.5+")
+@Since("INSERT VERSION")
+
+public class CondSmithingData extends PropertyCondition<Recipe> {
```

Comment:
this is probably better off just as a normal condition, since you're basically just circumventing everything property condition does anyway